ecommerce/api.py

```diff
@@ -472,3 +472,67 @@ def unenroll_learner_from_order(order_id):
                     )
             except ObjectDoesNotExist:
                 pass
+
+
+def check_pending_orders_for_resolution():
+    """Checks pending orders for resolution."""
+
+    gateway = PaymentGateway.get_gateway_class(ECOMMERCE_DEFAULT_PAYMENT_GATEWAY)
+
+    pending_orders = PendingOrder.objects.filter(
+        state=PendingOrder.STATE.PENDING
+    ).values_list("reference_number")
+
+    if len(pending_orders) == 0:
+        return
+
+    log.info(f"Resolving {len(pending_orders)} orders")
+
+    results = gateway.find_and_get_transactions(pending_orders)
+
+    if len(results.keys()) == 0:
+        log.info(f"No orders found to resolve.")
+        return
+
+    for result in results:
+        payload = results[result]
+        if int(payload["reason_code"]) == 100:
+
+            try:
+                order = PendingOrder.objects.filter(
+                    state=PendingOrder.STATE.PENDING,
+                    reference_number=payload["req_reference_number"],
+                ).get()
+
+                order.fulfill(payload)
+                order.save()
+
+                log.info(f"Fulfilled order {order.reference_number}.")
+            except Exception as e:
+                log.error(
+                    f"Couldn't process pending order for fulfillment {payload['req_reference_number']}: {str(e)}"
+                )
+        else:
+
+            try:
+                order = PendingOrder.objects.filter(
+                    state=PendingOrder.STATE.PENDING,
+                    reference_number=payload["req_reference_number"],
+                ).get()
+
+                order.cancel()
+                order.transactions.create(
```

Comment:
The logic here seems to duplicate the task and the command. I think we should introduce a common method that handles this thing and we call that function from within the loop here and in the command.

Reply:
I refactored this; I had them separate because the command did output during the process but info logs are fine I think. This also fixes the other issue you noted - I don't think it saved the changes it made. The API call one definitely does.